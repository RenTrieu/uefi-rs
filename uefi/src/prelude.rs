// SPDX-License-Identifier: MIT OR Apache-2.0

//! This module is used to simplify importing the most common UEFI types.
//!
//! This includes the system table modules, `Status` codes, etc.

pub use crate::{
    boot, cstr16, cstr8, entry, runtime, system, Handle, ResultExt, Status, StatusExt,
};
