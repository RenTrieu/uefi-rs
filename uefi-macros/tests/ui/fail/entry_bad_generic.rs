#![no_main]

use uefi::prelude::*;

#[entry]
fn main<T>() -> Status {
    Status::SUCCESS
}
