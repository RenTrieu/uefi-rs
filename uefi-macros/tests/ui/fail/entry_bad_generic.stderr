error: Entry function should not be generic
 --> tests/ui/fail/entry_bad_generic.rs:6:9
  |
6 | fn main<T>() -> Status {
  |         ^
